Dozer/DozerIcons.swift

```diff
@@ -255,9 +289,20 @@ public final class DozerIcons {
         }
     }
 
-    /// Determines if the user is interacting with the status bar based on level, owner and y-coordinate
+    /// hide and show dock icon and thus its menu bar: to free up space to show more menu bar icons
+    public class func toggleDockIcon(showIcon state: Bool) -> Bool {
+        var result: Bool
+        if state {
+            result = NSApp.setActivationPolicy(NSApplication.ActivationPolicy.regular)
+        } else {
+            result = NSApp.setActivationPolicy(NSApplication.ActivationPolicy.accessory)
+        }
+        return result
```

Comment:
I'll keep shuddering a little knowing that some code running on my machine is so devoted to not having early returns.

Reply:
Feel free to suggest a change

Reply:
Well, in this case I'd write it in one line:

`return NSApp.setActivationPolicy(state ? NSApplication.ActivationPolicy.regular : NSApplication.ActivationPolicy.accessory)`

Dunno if Swift's `[.thing]` syntax allows to remove the duplication of `NSApplication.ActivationPolicy` too.

Generally, I'm a big fan of early `return`s, especially for exceptional inputs that cause functions to not do usual stuff. They're pretty much saying “input wrong? no can do” right at the start, and as a bonus avoid extra nesting. Also they work smoothly with more-functional style, which is very nice, in my book.

Reply:
Swift warning
> Implicit Return Violation: Prefer implicit returns in closures, functions and getters. (implicit_return)

Reply:
Ah, apparently it's Swift [saying to drop `return` altogether in one-liner functions](https://github.com/apple/swift-evolution/blob/master/proposals/0255-omit-return.md). Didn't know about that. Seems they're bent even further in the functional direction than older mainstream languages.